<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="speed" default="1" />
  <arg name="randomize_path" default="false" />
  <arg name="align_with_middle_line" default="true" />
  <arg name="path_file" default="$(find gazebo_simulation)/param/automatic_drive/path.yaml" />

  <!-- Launch automatic drive test-->
  <node name="automatic_drive" ns="simulation" pkg="gazebo_simulation" type="automatic_drive_node" output="screen">
    <param name="speed" value="$(arg speed)" />
    <param name="randomize_path" value="$(arg randomize_path)" />
    <param name="align_with_middle_line" value="$(arg align_with_middle_line)" />

    <rosparam file="$(find gazebo_simulation)/param/automatic_drive/default.yaml" command="load" />
    <rosparam file="$(arg path_file)" command="load" />
    <rosparam file="$(find simulation_brain_link)/param/vehicle_simulation_link/default.yaml" command="load" ns="vehicle_simulation_link" />
    <rosparam file="$(find simulation_brain_link)/param/vehicle_simulation_link/topics.yaml" command="load" ns="topics/vehicle_simulation_link" />
    <rosparam file="$(find simulation_groundtruth)/param/groundtruth/topics.yaml" command="load" ns="topics/groundtruth" />
  </node>
</launch>
